dokka-subprojects/analysis-kotlin-symbols/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/symbols/kdoc/ResolveKDocLink.kt

```diff
@@ -48,11 +52,19 @@ internal fun KaSession.resolveKDocTextLinkToDRI(link: String, context: PsiElemen
  */
 internal fun KaSession.resolveKDocTextLinkToSymbol(link: String, context: PsiElement? = null): KaSymbol? {
     val kDocLink = createKDocLink(link, context)
-    return kDocLink?.let { resolveToSymbol(it) }
+    return kDocLink?.let { analyze(kDocLink) { resolveToSymbol(it) } }
 }
 
-private fun KaSession.createKDocLink(link: String, context: PsiElement? = null): KDocLink? {
-    val psiFactory = context?.let { KtPsiFactory.contextual(it) } ?: KtPsiFactory(this.useSiteModule.project)
+private fun KaSession.createKDocLink(link: String, context: PsiElement?): KDocLink? {
+    // see https://kotlin.github.io/analysis-api/in-memory-file-analysis.html
```

Comment:
TBH, the link doesn't help me to understand why we take a random file here. Maybe we can always enforce the `context` to be non-nullable on use-site and use ?
It would be also nice for me to understand why `KtPsiFactory(project)` doesn't work here

Reply:
Got it, yes, I mostly agree here, probably enforcing it will not have a lot of benefits.